### PR TITLE
XT-2948 - Create puppeteer JS config/smoke test for ixbrl-viewer

### PR DESCRIPTION
**Pull request by philipbalvanz-wf**
#### Reason for change
Additional browser based testing will help ensure overall product quality/compatibility.

#### Description of change
* Created a puppeteer config that runs in a github workflow against Window/MacOS/Ubuntu with multiple nodejs versions.  This workflow generates the viewer js, installs Arelle, generates a test project, then runs the supplied test against the generated file.   Should the test fail, it will store off a video of the run, as well as the generated viewers.
* Created a workflow/documentation for running the tests locally.
* There are compatibility issues with `puppeteer` and the version of `jest` defined in the repo, so I went ahead and bumped `jest` up to latest.

#### Steps to Test
* Github actions should include the new `Run Puppeteer Test` config in the various configurations
* I added instructions to the readme on how to run it locally, go ahead and give them a try to make sure they work.

**review**:
@Workiva/xt
@paulwarren-wk

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: tests/puppeteer/framework/page_objects/fact_details_panel.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        // Pull the concepts from the expected section
+        let conceptElems = await this.#viewerPage.page.$x(`
+            //*[contains(@class,"calculations")]//*[text()="${sectionTitle}"]
+            //ancestor::*[contains(@class,"card")]//*[contains(@class,"item")]
+        `);
+
+        let calc = {};
+        for(const elem in conceptElems){
+            let nameElem = await conceptElems[elem].waitForSelector('xpath/' + '*[contains(@class,"concept-name")]');
+            let name = await (await nameElem.getProperty('textContent')).jsonValue();
```
[the remark this answers is not in the document]
Comment (philipbalvanz-wf) [reply tests/puppeteer/framework/page_objects/fact_details_panel.js:64]: Can do!

File: tests/puppeteer/framework/viewer_page.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+export class ViewerPage {
+    browser;
+    page;
+    docFrame;
+    factDetailsPanel;
+
+    #artifactDirectory = './tests/puppeteer/artifacts';
+    #cleanedTestName = expect.getState().currentTestName.replaceAll(/[^a-zA-Z0-9-]/g, '_');
+    #isCi = process.env.CI || false;
```
[the remark this answers is not in the document]
Comment (philipbalvanz-wf) [reply tests/puppeteer/framework/viewer_page.js:15]: Will update.

**Review by sagesmith-wf — APPROVED** [review]

File: package.json
[the remark this answers is not in the document]
Comment (philipbalvanz-wf) [reply package.json:1]: Root doesn't have a package-lock.json?  Only the example directory has one.

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM

**Pull request merged**